packages/labs/ssr-dom-shim/src/lib/css-loader.ts

```diff
@@ -0,0 +1,42 @@
+import {readFileSync} from 'node:fs';
+import type {LoadHook, ResolveHook} from 'node:module';
+import {fileURLToPath} from 'node:url';
+
+/**
+ * Checks for each import, whether the file exists and if not, tries
+ * to find an associated TypeScript file.
+ *
+ * https://nodejs.org/api/module.html#resolvespecifier-context-nextresolve
+ */
+export const resolve: ResolveHook = (specifier, context, nextResolve) => {
+  if (context.importAttributes.type === 'css') {
+    return {
+      format: 'module',
+      shortCircuit: true,
+      url: new URL(specifier, context.parentURL).toString(),
+    };
+  }
+  return nextResolve(specifier, context);
+};
+
+/**
+ * When an attempt is made to import a CSS file/module, the CSS content
+ * is read and added to a CSSRule, which is provided in a CSSStyleSheet
+ * instance.
+ *
+ * https://nodejs.org/api/module.html#loadurl-context-nextload
+ */
+export const load: LoadHook = (url, context, nextLoad) => {
+  if (context.importAttributes.type === 'css') {
+    const filePath = fileURLToPath(url);
+    const css = readFileSync(filePath, 'utf8');
```

Comment:
fwiw, this would also work as well, since Node *fs** APIs support an instance of a URL, though maybe the constructor overhead isn't worth it?  But also one less NodeJS specific API I suppose (`fileURLToPath`) 🤷 
```js
const css = await fs.readFile(new URL(url), 'utf-8');
```

Reply:
I would prefer async fs APIs if they work. Will Node use the same thread for multiple imports if it yields?

Reply:
I'll change it as suggested.
However, it seems the future of hooks is synchronous: https://github.com/nodejs/node/issues/52219
But we can cross that bridge once that is widely available and/or the recommended approach.

Reply:
"Will Node use the same thread for multiple imports if it yields?"
My understanding is that Node.js runs hooks on a singular, separate thread.